src/main/java/favourite/drivers/FavouriteDatabaseAccess.java

```diff
@@ -0,0 +1,198 @@
+    // User entity in getUser method
+    private final UserReaderInterface userReader;
+    // partialpath is a String that contains the partial directory that leads to all csv files in the program
+    // it is isolated from the file name so that in case the file is moved, the code is still compatible
+    private final String partialPath;
+    /**
+     * Initializes DataAccess
+     *
+     * @param postReader an instance of PostReaderInterface
+     * @param userReader an instance of UserReaderInterface
+     * @param partialPath the partial path to database files
+     */
+    public FavouriteDatabaseAccess(PostReaderInterface postReader, UserReaderInterface userReader, String partialPath){
+        this.postReader = postReader;
+        this.userReader = userReader;
+        this.partialPath = partialPath;
+    }
+
+    /**
+     * Retrieves current user as a User object from the database
+     * @param userID the integer ID of the user being retrieved from the database
+     * @return the current user as a User object
+     */
+    @Override
+    public User getUser(int userID){
+        try {
+            // finding and retrieving the current user's data
+            List<String[]> allUsers = readAllLines(Paths.get(partialPath+"users.csv"));
+            String[] userData = new String[6];
+            for (String[] user : allUsers) {
+                try {
+                    int id = Integer.parseInt(user[0]);
+                    if (id == userID) {
+                        userData = user;
+                    }
+                } catch (NumberFormatException ex) {
+                    System.out.println("Reading the header of the csv");
+                }
+            }
+
+            // turning the current user's data into a user object
+            return userReader.readUser(userData);
+        }catch (IOException e1){
+            System.out.println("Error occurred while accessing file");
+        }catch (CsvException e2){
+            System.out.println("Error occurred while reading file");
+        }
+        return null;
+    }
+
+    /**
+     * Retrieves the post being favourited/unfavourited as a Post object from the database
+     *
+     * @param postID the integer ID of the post being favourited/unfavourited
+     * @return a Post object
+     */
+    @Override
+    public Post getPost(int postID){
+        try {
+            // finding and retrieving the current user's data
+            List<String[]> allPosts = readAllLines(Paths.get(partialPath+"posts.csv"));
+            String[] postData = new String[10];
+            for (String[] post : allPosts) {
+                try {
+                    int id = Integer.parseInt(post[0]);
+                    if (id == postID) {
+                        postData = post;
+                    }
+                } catch (NumberFormatException ex) {
+                    System.out.println("Reading the header of the csv");
```

Comment:
Same as above comment. I think it is a bit strange to print this every time you read the posts/user files.